pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -102,3 +170,15 @@ func (n Node) hasCompatibleResources(resourceList v1.ResourceList, it cloudprovi
 	}
 	return true
 }
+
+func (n *Node) recalculateDaemonResources() {
```

Comment:
My understanding is that this code intends to remove daemons from the overhead calculation once constraints are tightened. I'm not sure we want to do this for a couple of reasons:

1. I'm concerned about the performance impact of recomputing requirements on every loop
2. It's rare in practice for daemons to have large resource requests or complicated scheduling requirements 
3. Even with all of this additional math, we can't get it right in all cases. Consider the case where a daemonset only runs on arm64, but pods are compatible with either arm64 or amd64. We don't know the architecture until after the node is launched, at which point we can't change our binpacking decision. 
4. In the worst case scenario, we're creating nodes with additional room if it turns out the daemon can't schedule.

Instead, does it make sense to simply check the requirements once during `NewNode`. This will apply provisioner level requirements to the daemons, which is an easy to explain contract to customers. Further, if an edge use cases is causing undesirable behavior, the user has the ability to align the constraints of their provisioner and their daemonsets.

Reply:
SGTM, I implemented it because the old code had this behavior.  It didn't bin-pack until it had accumulated a set of the compatible pods, so it only did it once.  It's a separate commit on this PR as it requires removing an old test (from old hash bashed scheduling + bin-packing) that verified that this was performed.

I did go one step further and just computed daemon set resources given provisioner constraints as we only need to do it once that way.